src/node_crypto.cc

```diff
@@ -1689,10 +1711,53 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
         reinterpret_cast<unsigned char*>(Buffer::Data(pubbuff));
     i2d_RSA_PUBKEY(rsa.get(), &pubserialized);
     info->Set(env->context(), env->pubkey_string(), pubbuff).FromJust();
+  } else if (ec) {
+    const EC_GROUP* group = EC_KEY_get0_group(ec.get());
+    if (group != nullptr) {
+      int bits = EC_GROUP_order_bits(group);
+      if (bits > 0) {
+        info->Set(context, env->bits_string(),
+                  Integer::New(env->isolate(), bits)).FromJust();
+      }
+    }
+
+    unsigned char* pub = nullptr;
+    size_t publen = EC_KEY_key2buf(ec.get(), EC_KEY_get_conv_form(ec.get()),
+                                   &pub, nullptr);
+    if (publen > 0) {
+      Local<Object> buf = Buffer::New(env, pub, publen).ToLocalChecked();
+      // Ownership of pub pointer accepted by Buffer.
+      pub = nullptr;
+      info->Set(context, env->pubkey_string(), buf).FromJust();
+    } else {
+      CHECK_NULL(pub);
+    }
+
+    if (EC_GROUP_get_asn1_flag(group) != 0) {
```

Comment:
I think this check for `asn1_flag` is unnecessary? It should be enough to just check `nid != 0`—if the curve has a name, `nid` will be non-zero, and if it doesn't, `nid` will be zero. No need to also check the `asn1_flag`, right?

(The reason I'm asking is because `EC_GROUP_get_asn1_flag` doesn't exist in BoringSSL and I'm working on BoringSSL integration downstream in Electron.)

Reply:
The code was directly copied from https://github.com/nodejs/node/blob/9dfbc391fafe0495d30512ab811d4e372f9be449/deps/openssl/openssl/crypto/ec/eck_prn.c#L97-L119  but reading through the docs, yes, I suppose just trying to get the nid would work. I'll update this.

Reply:
Thanks!

Reply:
see: https://github.com/nodejs/node/pull/25345